bin/install_dependencies.ps1

```diff
@@ -10,36 +10,49 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
-# requires English language pack installed
-choco install visualstudio2017buildtools "--passive --locale en-US"
-choco install visualstudio2017-workload-vctools --package-parameters "--includeRecommended --add Microsoft.VisualStudio.Component.VC.ATLMFC"
-choco install visualstudio2017-workload-nativedesktop
-choco install windows-sdk-10.1 nodejs-lts wget nasm cyg-get wixtoolset python3 make nssm gpg4win checksum archiver dependencywalker unzip
-choco install nsis --version=2.51
+# Sourcing variable definitions
+. ${PSScriptRoot}\variables.ps1
+
+# Exclude c:\relax from MS defender to speed up things
+Add-MpPreference -ExclusionPath "C:\relax"
+
+# Install build tools - requires English language pack installed
+choco install visualstudio2022buildtools "--passive --locale en-US"
+choco install visualstudio2022-workload-vctools --package-parameters "--add Microsoft.VisualStudio.Component.VC.ATL --add Microsoft.VisualStudio.Component.VC.Redist.MSM --add Microsoft.Net.Component.4.8.TargetingPack"
+choco install nodejs-lts wixtoolset make nssm python3 vswhere gnuwin32-coreutils.portable
```

Comment:
One other improvement I was thinking of to help when we are troubleshooting or trying to reproduce past builds, was adding info on how to dump version info of what packages have been installed by choco, vcpkg, etc, and a suggested place to store this for the convenience binaries.

Reply:
Yes, I think this is a good idea! We should discuss this in detail.
For the short:
1. `choco list --local-only -v`gives the local packages with version number.
2. For VCPKG: maybe the git hash and [this](https://github.com/microsoft/vcpkg/blob/master/docs/examples/versioning.getting-started.md) helps.